share/completions/pre-commit.fish

```diff
@@ -0,0 +1,90 @@
+set -l commands autoupdate clean gc init-templatedir install install-hooks migrate-config run sample-config try-repo uninstall validate-config validate-manifest
+set -l hook_stages commit merge-commit prepare-commit-msg commit-msg post-commit manual post-checkout push post-merge post-rewrite
+set -l hook_types pre-commit pre-merge-commit pre-push prepare-commit-msg commit-msg post-commit post-checkout post-merge post-rewrite
+
+functions -q __fish_git || source $__fish_data_dir/completions/git.fish
+
+function __fish_pre_commit_config_print -a key
+    test -r .pre-commit-config.yaml && string match -rg "\s+$key:\s+(\S+)" <.pre-commit-config.yaml | string unescape
+end
```

Comment:
Not sure if we can do better, but this parses the [yaml config file](https://pre-commit.com/#2-add-a-pre-commit-configuration) while trying to tolerant comments and different quoting styles like:

```yaml
- id: unquoted # comment
- id: 'single-quoted'
- id: "double-quoted"
```

Reply:
I don't think I know anything, so let's just keep it like that for now.